### PR TITLE
chore/linting

### LINKED ISSUE #23: CI sec check fail & no linting is present
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The CI checks regarding security fail & no linting is done on the entirety of the repository. This may result in poorly maintained code and errors which go unchecked.

### Expected Behavior

We have automatic linting to avoid typical code mistakes & code smells.

### Steps To Reproduce

_No response_

### Relevant logs and/or screenshots, environment information, etc.

_No response_

### Who can address the issue?

Everyone

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by puffitos**
## Motivation

Close #23 

## Changes

Added linting with golangci, turned off some of the most strict linters. Also implemented most of the linter's suggestions. We should discuss some of the TODOs before merging (a lot of errors were left unchecked...)

## TODO

- [x] I've assigned this PR to myself
- [x] I've labeled this PR correctly

### CODE REVIEW

File: pkg/checks/health.go [hunk 1]
```diff
@@ -82,32 +82,32 @@ func (h *Health) Run(ctx context.Context) error {
...
+func (h *Health) Shutdown(_ context.Context) error {
 	http.Handle(h.route, http.NotFoundHandler())
 	h.done <- true
 
 	return nil
 }
 
 // SetConfig sets the configuration for the health check
-func (h *Health) SetConfig(ctx context.Context, config any) error {
+func (h *Health) SetConfig(_ context.Context, config any) error {
```
Comment (lvlcn-t) [inline pkg/checks/health.go:110]: I'd either use the context for logging or smth or completely remove it.

File: pkg/checks/health.go [hunk 2]
```diff
@@ -120,35 +120,39 @@ func (h *Health) SetConfig(ctx context.Context, config any) error {
...
+				log.Error("Could not write response", "error", err.Error())
+				// TODO: Discuss what should happen
+			}
 		})
 	}
 }
 
 // DeregisterHandler dynamically deletes the server handler
-func (h *Health) DeregisterHandler(ctx context.Context, router *api.RoutingTree) {
+func (h *Health) DeregisterHandler(_ context.Context, router *api.RoutingTree) {
```
Comment (lvlcn-t) [inline pkg/checks/health.go:141]: > I'd either use the context for logging or smth or completely remove it.

same here

File: pkg/checks/health.go [hunk 3]
```diff
@@ -200,7 +204,7 @@ func getHealth(ctx context.Context, url string) error {
 		Timeout: time.Second * 5,
 	}
 
-	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	req, err := http.NewRequestWithContext(ctx, "GET", url, http.NoBody)
```
Comment (lvlcn-t) [inline pkg/checks/health.go:207]: Also use `http.MethodGet` instead of hardcoded "GET"

File: pkg/config/http.go [hunk 4]
```diff
@@ -87,7 +86,7 @@ func (gl *HttpLoader) GetRuntimeConfig(ctx context.Context) (*RuntimeConfig, err
 	client := http.DefaultClient
 	client.Timeout = gl.cfg.Loader.http.timeout
 
-	req, err := http.NewRequestWithContext(ctx, "GET", gl.cfg.Loader.http.url, nil)
+	req, err := http.NewRequestWithContext(ctx, "GET", gl.cfg.Loader.http.url, http.NoBody)
```
Comment (lvlcn-t) [inline pkg/config/http.go:89]: > Also use `http.MethodGet` instead of hardcoded "GET"

same here

File: pkg/sparrow/run.go [hunk 5]
```diff
@@ -131,25 +138,37 @@ func (s *Sparrow) ReconcileChecks(ctx context.Context) {
...
+		err := existingCheck.Shutdown(ctx)
+		if err != nil {
+			log.ErrorContext(ctx, "Failed to shutdown check", "error", err)
+			// TODO: discuss what should happen
+		}
+		if c, ok := s.resultFanIn[existingCheckName]; ok {
+			// close fan in the channel if it exists
+			close(c)
+			delete(s.resultFanIn, existingCheckName)
 		}
```
Comment (lvlcn-t) [inline pkg/sparrow/run.go:168]: In my opinion these variable names are still way too long, why not `name`, `check` or even `n`, `c`?

**Review by lvlcn-t — COMMENTED** [review]
Some minor things that can be improved

**Comment by lvlcn-t** [comment]
Additionally we should talk about/take care of the many open TODO comments that are in our code base.

File: pkg/sparrow/run.go (hunk 5 above)
Comment (puffitos) [reply pkg/sparrow/run.go]: You've refactored the function so we'll tackle that there

**Comment by puffitos** [comment]
PR ready after call with developers. The tests will fail (errors unhandled and unittests). Those are handled in a separate PR.

File: pkg/config/http.go [hunk 6]
```diff
@@ -70,7 +69,7 @@ func (gl *HttpLoader) Run(ctx context.Context) {
 		gl.cCfgChecks <- runtimeCfg.Checks
 
 		timer := time.NewTimer(gl.cfg.Loader.Interval)
-		defer timer.Stop()
+		defer timer.Stop() //nolint:gocritic // TODO: check if this is right
```
Comment (lvlcn-t) [inline pkg/config/http.go:72]: This will be part of another PR for #35?

**Review by lvlcn-t — APPROVED** [review]
LGTM overall, just had some minor questions/notes.

File: pkg/config/http.go (hunk 6 above)
Comment (y-eight) [reply pkg/config/http.go:72]: right!

**Comment by y-eight** [comment]
I have just added 2 small things: Indent for json response of checks metrics endpoint; Reduced interval of health check to 15 sec. This will be configurable in the future by  https://github.com/caas-team/sparrow/issues/34

PR is LGTM

File: internal/helper/retry.go [hunk 7]
```diff
@@ -49,7 +49,7 @@ func Retry(effector Effector, rc RetryConfig) func(ctx context.Context) error {
 			log.WarnContext(ctx, fmt.Sprintf("Effector call failed, retrying in %v", delay))
 
 			timer := time.NewTimer(delay)
-			defer timer.Stop()
+			defer timer.Stop() //nolint:gocritic //TODO: check if this is correct
```
[the remark this answers is not in the document]
Comment (y-eight) [reply internal/helper/retry.go:52]: rigth!

**Review by y-eight — APPROVED** [review]
